lib/src/migrators/module/scope.dart

```diff
@@ -0,0 +1,140 @@
+// Copyright 2019 Google LLC
+//
+// Use of this source code is governed by an MIT-style
+// license that can be found in the LICENSE file or at
+// https://opensource.org/licenses/MIT.
+
+import 'package:sass_migrator/src/utils.dart';
+
+// The sass package's API is not necessarily stable. It is being imported with
+// the Sass team's explicit knowledge and approval. See
+// https://github.com/sass/dart-sass/issues/236.
+import 'package:sass/src/ast/sass.dart';
+
+import 'package:path/path.dart' as p;
+import 'package:source_span/source_span.dart';
+
+/// Keeps track of the scope of any members declared at the current level of
+/// the stylesheet.
+class Scope {
+  /// The parent of this scope, or null if this scope is global.
+  final Scope parent;
+
+  /// Variables defined in this scope.
+  ///
+  /// These are usually VariableDeclarations, but can also be Arguments from
+  /// a CallableDeclaration.
+  final variables = normalizedMap<SassNode /*VariableDeclaration|Argument*/ >();
+
+  /// Mixins defined in this scope.
+  final mixins = normalizedMap<MixinRule>();
+
+  /// Functions defined in this scope.
+  final functions = normalizedMap<FunctionRule>();
+
+  /// Nested imports from this scope.
+  ///
+  /// Note: For the purposes of migration, any imports that come after a rule
+  /// other than @import, @use, @forward, or @charset are considered nested, since
+  /// they can't be migrated to @use.
+  final nestedImports = <Uri>{};
+
+  Scope(this.parent);
+
+  /// The global scope this scope descends from.
+  Scope get global => parent?.global ?? this;
+
+  /// Returns whether a variable [name] exists in a non-global scope.
+  bool isLocalVariable(String name) =>
+      parent != null &&
+      (variables.containsKey(name) || parent.isLocalVariable(name));
+
+  /// Returns whether a mixin [name] exists in a non-global scope.
+  bool isLocalMixin(String name) =>
+      parent != null && (mixins.containsKey(name) || parent.isLocalMixin(name));
+
+  /// Returns whether a function [name] exists in a non-global scope.
+  bool isLocalFunction(String name) =>
+      parent != null &&
+      (functions.containsKey(name) || parent.isLocalFunction(name));
+
+  /// Constructs and throws a MigrationException for a reference to a member
+  /// from a nested import.
+  void _errorNestedImport(String type, Uri importUrl, FileSpan reference) {
```

Comment:
Explain this in more detail. We can expect that most users of the migrator aren't going to be intimately familiar with the module system, so they may not understand that nested imports get translated to the `load-css()` mixin.

Maybe something like "This $type was loaded from a nested import. The module system only supports loading nested CSS using the load-css() function, which doesn't load $types."